google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/JobSnippets.java

```diff
@@ -48,4 +53,60 @@ public boolean isDone() {
     }
     return true;
   }
+
+  /**
+   * Example usage of {@code waitFor()}.
+   */
+  // [TARGET waitFor(WaitForOption...)]
+  public boolean waitFor() {
+    try {
+      // [START waitFor]
+      Job completedJob = job.waitFor();
+      if (completedJob == null) {
+        // job no longer exists
+      } else if (completedJob.status().error() != null) {
+        // job failed, handle error
+      } else {
+        // job completed successfully
+      }
+      // [END waitFor]
+    } catch (InterruptedException e) {
+      return true;
```

Comment:
Mostly because I didn't know what to do with InterruptedException and I saw that waitFor documents that it can throw it. I've changed the snippets to throw InterruptedException instead based on the recommendations here: https://www.ibm.com/developerworks/library/j-jtp05236/